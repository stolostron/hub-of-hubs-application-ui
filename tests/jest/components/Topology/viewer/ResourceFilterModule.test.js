// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
"use strict";

import {
  ResourceFilterView,
  ResourceFilterModule
} from "../../../../../src-web/components/Topology/viewer/ResourceFilterModule";
import renderer from "react-test-renderer";
import React from "react";

const mockViewData = {
  activeFilters: {
    resourceStatuses: new Set(["red", "orange", "yellow"]),
    type: [
      "application",
      "cluster",
      "placements",
      "service",
      "statefulset",
      "storageclass",
      "subscription"
    ]
  },
  availableFilters: {
    namespaces: {
      name: "Namespaces",
      availableSet: new Set(["cassandra-ns", "cluster-scoped"])
    },
    resourceStatuses: {
      name: "Resource status",
      availableSet: new Map([
        ["green", "Success"],
        ["orange", "Pending"],
        ["yellow", "Warning"],
        ["red", "Error"]
      ])
    },
    resourceTypes: {
      name: "Types",
      availableSet: new Set(["statefulset", "service", "storageclass"])
    },
    type: [
      "application",
      "cluster",
      "placements",
      "service",
      "statefulset",
      "storageclass",
      "subscription"
    ]
  },
  locale: "en-US",
  onClose: jest.fn(),
  updateFilters: jest.fn()
};

const mockModuleData = {
  activeFilters: {
    resourceStatuses: new Set(["red", "orange", "yellow"]),
    type: [
      "application",
      "cluster",
      "placements",
      "service",
      "statefulset",
      "storageclass",
      "subscription"
    ]
  },
  availableFilters: {
    namespaces: {
      name: "Namespaces",
      availableSet: new Set(["cassandra-ns", "cluster-scoped"])
    },
    resourceStatuses: {
      name: "Resource status",
      availableSet: new Map([
        ["green", "Success"],
        ["orange", "Pending"],
        ["yellow", "Warning"],
        ["red", "Error"]
      ])
    },
    resourceTypes: {
      name: "Types",
      availableSet: new Set(["statefulset", "service", "storageclass"])
    },
    type: [
      "application",
      "cluster",
      "placements",
      "service",
      "statefulset",
      "storageclass",
      "subscription"
    ]
  },
  locale: "en-US",
  portals: {
    assortedFilterOpenBtn: "assorted-filter-open-portal-id",
    assortedFilterCloseBtns: "assorted-filter-close-portal-id",
    typeFilterBar: "type-filter-bar-portal-id",
    searchTextbox: "search-textbox-portal-id"
  },
  updateActiveFilters: jest.fn()
};

describe("Test ResourceFilterView renderView", () => {
  it("should render the html", () => {
    const rfv = new ResourceFilterView();

    rfv.renderView({ style: {}, props: {} });
  });
});

describe("ResourceFilterView", () => {
  it("renders as expected", () => {
    const component = renderer.create(
      <ResourceFilterView
        activeFilters={mockViewData.activeFilters}
        availableFilters={mockViewData.availableFilters}
        locale={mockViewData.locale}
        onClose={mockViewData.onClose}
        updateFilters={mockViewData.updateFilters}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("ResourceFilterModule", () => {
  it("renders as expected", () => {
    const component = renderer.create(
      <ResourceFilterModule
        activeFilters={mockModuleData.activeFilters}
        availableFilters={mockModuleData.availableFilters}
        locale={mockModuleData.locale}
        portals={mockModuleData.portals}
        updateActiveFilters={mockModuleData.updateActiveFilters}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
