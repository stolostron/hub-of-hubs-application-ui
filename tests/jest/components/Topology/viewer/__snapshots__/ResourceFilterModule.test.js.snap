// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceFilterModule renders as expected 1`] = `
<div
  className="resource-filter-button"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <svg
    className="button-icon"
  >
    <use
      href="#diagramIcons_filter"
    />
  </svg>
  <div
    className="button-label"
  >
    Filter
  </div>
</div>
`;

exports[`ResourceFilterView renders as expected 1`] = `
<div
  className="resource-filter-view"
  style={
    Object {
      "height": 771,
    }
  }
>
  <h3
    className="filterHeader"
  >
    <svg
      className="titleIcon"
    >
      <use
        href="#diagramIcons_filter"
      />
    </svg>
    <div
      className="titleText"
    >
      Filters
    </div>
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      className="closeIcon"
      description="Close filter view"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 352 512"
      width="1em"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
      />
    </svg>
  </h3>
  <div
    className="filter-sections-container"
    style={
      Object {
        "height": 708,
        "overflow": "hidden",
        "position": "relative",
        "width": 260,
      }
    }
  >
    <div
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "bottom": 0,
          "left": 0,
          "marginBottom": 0,
          "marginRight": 0,
          "overflow": "scroll",
          "overflowX": "hidden",
          "overflowY": "scroll",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <div
        className="filter-section"
      >
        <div
          className="filter-section-title"
        >
          Namespaces
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={true}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-1"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="namespacesall"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="namespacesall"
          >
            All
          </label>
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={false}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-2"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="namespacescassandra-ns"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="namespacescassandra-ns"
          >
            cassandra-ns
          </label>
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={false}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-3"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="namespacescluster-scoped"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="namespacescluster-scoped"
          >
            cluster-scoped
          </label>
        </div>
      </div>
      <div
        className="filter-section"
      >
        <div
          className="filter-section-title"
        >
          Resource status
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={false}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-4"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="resourceStatusesall"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="resourceStatusesall"
          >
            All
          </label>
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={true}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-5"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="resourceStatusesred"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="resourceStatusesred"
          >
            Error
          </label>
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={true}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-6"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="resourceStatusesorange"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="resourceStatusesorange"
          >
            Pending
          </label>
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={false}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-7"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="resourceStatusesgreen"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="resourceStatusesgreen"
          >
            Success
          </label>
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={true}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-8"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="resourceStatusesyellow"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="resourceStatusesyellow"
          >
            Warning
          </label>
        </div>
      </div>
      <div
        className="filter-section"
      >
        <div
          className="filter-section-title"
        >
          Types
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={true}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-9"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="resourceTypesall"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="resourceTypesall"
          >
            All
          </label>
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={false}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-10"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="resourceTypesservice"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="resourceTypesservice"
          >
            service
          </label>
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={false}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-11"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="resourceTypesstatefulset"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="resourceTypesstatefulset"
          >
            statefulset
          </label>
        </div>
        <div
          className="pf-c-check filter-section-checkbox"
        >
          <input
            aria-invalid={false}
            checked={false}
            className="pf-c-check__input"
            data-ouia-component-id="OUIA-Generated-Checkbox-12"
            data-ouia-component-type="PF4/Checkbox"
            data-ouia-safe={true}
            disabled={false}
            id="resourceTypesstorageclass"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="pf-c-check__label"
            htmlFor="resourceTypesstorageclass"
          >
            storageclass
          </label>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "height": 6,
          "left": 2,
          "position": "absolute",
          "right": 2,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "height": "100%",
            "position": "relative",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "position": "absolute",
          "right": 2,
          "top": 2,
          "width": 6,
        }
      }
    >
      <div
        className="filter-sections-scrollbar"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "position": "relative",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
